src/lib/blkalloc/append_blk_allocator.h

```diff
@@ -97,8 +102,8 @@ class AppendBlkAllocator : public BlkAllocator {
     void on_meta_blk_found(const sisl::byte_view& buf, void* meta_cookie);
 
 private:
+    std::mutex m_mtx; // thread_safe, TODO: open option for consumer to choose to go lockless;
```

Comment:
folly::RWSpinLock?

Reply:
should be fine for either of them since they are all in-mem ops.

Reply:
I would say to keep with mutex to give other threads some cycles to run to achieve better throughput. 
It is only a few cpu cycles anyway within the scope of this mutex. We are concurrently writing the same chunks so I/O latency would be the overwhelming factor of the total latency.